airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_component_schema.yaml

```diff
@@ -1010,6 +1010,81 @@ definitions:
       $parameters:
         type: object
         additionalProperties: true
+  SessionTokenAuthenticator:
+    type: object
+    required:
+      - type
+      - login_requester
+      - session_token_path
+      - request_authentication
+    properties:
+      type:
+        type: string
+        enum: [SessionTokenAuthenticator]
+      login_requester:
+        title: Login Requester
+        description: Description of the request to perform to obtain a session token to perform data requests. The response body is expected to be a JSON object with a session token property.
+        "$ref": "#/definitions/HttpRequester"
+        examples:
+          - type: HttpRequester
+            url_base: "https://my_api.com"
+            path: "/login"
+            authenticator:
+              type: BasicHttpAuthenticator
+              username: "{{ config.username }}"
+              password: "{{ config.password }}"
+      session_token_path:
+        title: Session Token Path
+        description: The path in the response body returned from the login requester to the session token.
+        examples:
+          - ["access_token"]
+          - ["result", "token"]
+        type: array
+        items:
+          type: string
+      expiration_time:
```

Comment:
nit: can we call this `expiration_duration`, `expires_in`, or something else that makes it clear this is a duration and not a timestamp